functional/FunctionalTest.java

```diff
@@ -2410,8 +2414,8 @@ public static void testSetBucketPolicy(String objectPrefix, PolicyType policyTyp
     long startTime = System.currentTimeMillis();
     try {
       client.setBucketPolicy(bucketName, objectPrefix, policyType);
-      // Wait for 5 seconds for server to set the policy into effect.
-      Thread.sleep(5000);
+      // Wait for 15 seconds for server to set the policy into effect.
+      Thread.sleep(15000);
```

Comment:
Have you checked 15 seconds wait is minimum required?

Reply:
The minimum required seems to be around 7 seconds, so doubled that number.